src/compile/split.ts

```diff
@@ -29,7 +29,7 @@ export class Split<T extends Object> {
     return this.explicit[key] !== undefined ? this.explicit[key] : this.implicit[key];
   }
 
-  public getWithType<K extends keyof T>(key: K): {explicit: boolean, value: T[K]} {
+  public getWithExplicit<K extends keyof T>(key: K): Explicit<T[K]> {
```

Comment:
get the particular value of the given key with its `explicit` metadata so I prefer `getWithExplicit`.